adb_graphics/datahandler/gribdata.py

```diff
@@ -559,9 +567,14 @@ def fire_weather_index(self, values, **kwargs) -> np.ndarray:
                 (mois ** 13.55) * \
                 snowc)
 
+        temp.close()
+        dewpt.close()
+        weasd.close()
+        gust.close()
+        soilm.close()
+
         return fwi
 
-    @property
```

Comment:
I think that object properties also save the data structure in memory until the object is deleted, so this was another way to allow the freeing of memory.